File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/exchanges/ExchangePlayerMultiCraftRequestMessage.scala





















// Generated on 05/03/2013 10:34:55
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ExchangePlayerMultiCraftRequestMessage extends ExchangeRequestMessage {
  override val messageId = 5784

  protected trait Message extends ExchangeRequestMessage.Message with Serializable {
    override val messageId = ExchangePlayerMultiCraftRequestMessage.messageId

    val target: Int
    val skillId: Int

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeInt(target)
      buf.writeInt(skillId)
    }

  }

  protected def create(exchangeType: Byte, target: Int, skillId: Int): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val exchangeType = buf.readByte()
    val target = buf.readInt()
    val skillId = buf.readInt()

    create(exchangeType, target, skillId)
  }

}

object ExchangePlayerMultiCraftRequestMessage extends ExchangePlayerMultiCraftRequestMessage {
  case class MessageImpl(exchangeType: Byte, target: Int, skillId: Int) extends Message
  type Target = MessageImpl

  override def create(exchangeType: Byte, target: Int, skillId: Int) = MessageImpl(exchangeType, target, skillId)

  def apply(exchangeType: Byte, target: Int, skillId: Int) = create(exchangeType, target, skillId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ExchangePlayerMultiCraftRequestMessage.java


// Generated on 05/08/2013 19:37:54
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ExchangePlayerMultiCraftRequestMessage extends ExchangeRequestMessage {
    public static final int MESSAGE_ID = 5784;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int target;
    public int skillId;
    
    public ExchangePlayerMultiCraftRequestMessage() { }
    
    public ExchangePlayerMultiCraftRequestMessage(byte exchangeType, int target, int skillId) {
        super(exchangeType);
        this.target = target;
        this.skillId = skillId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeInt(target);
        buf.writeInt(skillId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        target = buf.readInt();
        if (target < 0)
            throw new RuntimeException("Forbidden value on target = " + target + ", it doesn't respect the following condition : target < 0");
        skillId = buf.readInt();
        if (skillId < 0)
            throw new RuntimeException("Forbidden value on skillId = " + skillId + ", it doesn't respect the following condition : skillId < 0");
    }
    
}
